src/scheduler.js

```diff
@@ -162,8 +162,14 @@ class Scheduler extends events.EventEmitter {
       errJson = `(error formatting JSON: ${e})`;
     }
 
-    this.notify.email({
-      address: email,
+    let email = this.createEmail(hook, err, errJson);
+    this.notify.email(email);
+    this.notify.lastEmail = _.merge({}, email, {err, errJson});
```

Comment:
I like that you have factored out creating the email!

However, I think that the fake `email` method should set `lastEmail`, rather than setting it here.  This is "production" (meaning, not test) code, so where possible it's best to not write testing-specific things here.